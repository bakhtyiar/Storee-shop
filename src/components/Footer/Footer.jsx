import React from 'react';
import {Container} from "react-bootstrap";

const Footer = () => {
    return (
        <Container fluid style={{'backgroundColor': '#212529', 'color': '#ececec', 'padding': '48px 0', 'display': 'flex', 'flex': '0 0 auto'}}>
            <Container style={{'display': 'flex', 'justify-content': 'center'}}>
                Copyright Â© 3000, All Rights Reserved
            </Container>
        </Container>
    );
};

export default Footer;
