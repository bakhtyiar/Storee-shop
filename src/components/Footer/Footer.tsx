import React, {useContext} from 'react';
import {Container, Form} from "react-bootstrap";
import {RootContext} from "../../contexts/root-context/root-context";
import {capitalizeStr} from "../../utils/str/str";

const Footer = () => {
    const {themeState: {theme, onSwitchTheme}} = useContext(RootContext);

    return (
        <Container
            fluid
            className='bg-dark text-light mt-3 py-3 px-0 d-flex flex-grow-0 flex-shrink-0 '
            data-testid='footer'
        >
            
            <Container className='d-flex justify-content-between'>
                <span>Copyright Â© 3000, All Rights Reserved</span>
                <Form.Check
                    reverse
                    type="switch"
                    id="theme-switch"
                    label={`${capitalizeStr(theme)} theme`}
                    onChange={onSwitchTheme}
                />
            </Container>
        </Container>
    );
};

export default Footer;
