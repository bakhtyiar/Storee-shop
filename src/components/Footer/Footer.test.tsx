import React from "react";
import {render, screen} from "@testing-library/react";
import Footer from "./Footer";

describe("Basic tests", () => {
    test("Match snapshot", () => {
        
        render(<Footer/>);
        expect(screen.getByTestId("footer")).toMatchSnapshot();
    })
})