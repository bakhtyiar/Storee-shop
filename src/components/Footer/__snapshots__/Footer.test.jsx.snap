// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic tests Match snapshot 1`] = `
<div
  class="bg-dark text-light mt-3 py-3 px-0 d-flex flex-grow-0 flex-shrink-0  container-fluid"
  data-testid="footer"
>
  <div
    class="d-flex justify-content-center container"
  >
    Copyrighasdt Â© 3000, All Rights Reserved
  </div>
</div>
`;
